### PR TITLE
Introduce new rule sssd_ldap_configure_tls_reqcert

### PR DESCRIPTION
**Pull request by ggbecker**
#### Description:

- This pull request depends on changes from #6032 and only a0bdbcb is relevant to this PR, until the other one gets merged, more changes will show up here.

- Introduce new rule `sssd_ldap_configure_tls_reqcert`

#### Rationale:

Reference: https://vaulted.io/library/disa-stigs-srgs/red_hat_enterprise_linux_7_security_technical_implementation_guide/V-72229?version=v2r7

### CODE REVIEW

**Comment by mildas** [comment]
**Changes identified:**
Rule sssd_ldap_configure_tls_ca_dir:
 Found change in bash remediation.
 Templatization usage changed.
Rule sssd_ldap_configure_tls_reqcert:
 Templatization usage changed.
 OVAL check is newly added.
 Ansible remediation newly added.
 Bash remediation is newly added.
Rule sssd_ldap_start_tls:
 Node deleted from OVAL check.
 Found change in bash remediation.
 Node moved within OVAL check.
 Deleted attribute from OVAL check.
 Text changed in OVAL check.
 New node inserted to OVAL check.
 Attribute value changed in OVAL check.
 Text added outsite tags in OVAL check.
 Templatization usage changed.
Rule checks:
 The rule doesn't occur in any profile nor product.
 OVAL check is newly added.
Profile stig on rhel7:
 Rule sssd_ldap_configure_tls_reqcert added to stig profile.
Macro ansible_sssd_ldap_config:
 In Ansible remediation for sssd_ldap_start_tls.
 In Ansible remediation for sssd_ldap_configure_tls_ca_dir.
 In Ansible remediation for sssd_ldap_configure_tls_reqcert.
Macro bash_sssd_ldap_config:
 In Bash remediation for sssd_ldap_configure_tls_ca_dir.
 In Bash remediation for sssd_ldap_start_tls.
 In Bash remediation for sssd_ldap_configure_tls_reqcert.
Others:
 Python abstract syntax tree change found in ssg/constants.py.

**Recommended tests to execute:**
 build_product rhel7
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using ansible --datastream build/ssg-rhel7-ds.xml sssd_ldap_configure_tls_reqcert
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-rhel7-ds.xml sssd_ldap_configure_tls_reqcert
 tests/test_suite.py profile --libvirt qemu:///system test-suite-vm --datastream build/ssg-rhel7-ds.xml stig
 (cd build && cmake ../ && ctest -j4)
 build_product ol7
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using ansible --datastream build/ssg-ol7-ds.xml sssd_ldap_configure_tls_ca_dir
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-ol7-ds.xml sssd_ldap_configure_tls_ca_dir
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using ansible --datastream build/ssg-ol7-ds.xml sssd_ldap_start_tls
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-ol7-ds.xml sssd_ldap_start_tls

**Comment by redhatrises** [comment]
@ggbecker there are files in here that appear to be part of #6032. Can you remove them?

**Comment by ggbecker** [comment]
> @ggbecker there are files in here that appear to be part of #6032. Can you remove them?

No, because this pull request depends on changes from #6032. As soon as this other pull request gets merged, these `extra` changes will disappear in this pull request. This pull requests uses jinja macros introduced in #6032

**Comment by mildas** [comment]
**Changes identified:**
Rule sssd_ldap_configure_tls_reqcert:
 Bash remediation is newly added.
 Ansible remediation newly added.
 OVAL check is newly added.
Profile stig on rhel7:
 Rule sssd_ldap_configure_tls_reqcert added to stig profile.

**Recommended tests to execute:**
 build_product rhel7
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using ansible --datastream build/ssg-rhel7-ds.xml sssd_ldap_configure_tls_reqcert
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-rhel7-ds.xml sssd_ldap_configure_tls_reqcert
 tests/test_suite.py profile --libvirt qemu:///system test-suite-vm --datastream build/ssg-rhel7-ds.xml stig

**Comment by mildas** [comment]
**Changes identified:**
Rule sssd_ldap_configure_tls_reqcert:
 Ansible remediation newly added.
 Bash remediation is newly added.
 OVAL check is newly added.
Profile stig on rhel7:
 Rule sssd_ldap_configure_tls_reqcert added to stig profile.

**Recommended tests to execute:**
 build_product rhel7
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using ansible --datastream build/ssg-rhel7-ds.xml sssd_ldap_configure_tls_reqcert
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-rhel7-ds.xml sssd_ldap_configure_tls_reqcert
 tests/test_suite.py profile --libvirt qemu:///system test-suite-vm --datastream build/ssg-rhel7-ds.xml stig

**Review by jan-cerny — COMMENTED** [review]
It looks great. I checked that the tests scenarios are passing. I have only 3 nitpicks.

**Comment by jan-cerny** [comment]
@ggbecker Do you have any plans?

**Comment by ggbecker** [comment]
> @ggbecker Do you have any plans?

Working on that right now. Thanks

File: linux_os/guide/services/sssd/sssd-ldap/sssd_ldap_configure_tls_reqcert/rule.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+documentation_complete: true
+
+prodtype: ol7,ol8,rhel7,rhel8,wrlinux1019
+
+title: 'Configure SSSD LDAP Backend Client to Demand a Valid Certificate from the Server'
+
+description: |-
+    Configure SSSD to demand a valid certificate from the server to
+    protect the integrity of LDAP remote access sessions. By setting
```
[the remark this answers is not in the document]
Comment (ggbecker) [reply linux_os/guide/services/sssd/sssd-ldap/sssd_ldap_configure_tls_reqcert/rule.yml]: Done

File: linux_os/guide/services/sssd/sssd-ldap/sssd_ldap_configure_tls_reqcert/oval/shared.xml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+<def-group>
+  <definition class="compliance" id="sssd_ldap_configure_tls_reqcert" version="1">
+    <metadata>
```
[the remark this answers is not in the document]
Comment (ggbecker) [reply linux_os/guide/services/sssd/sssd-ldap/sssd_ldap_configure_tls_reqcert/oval/shared.xml]: Done

File: linux_os/guide/services/sssd/sssd-ldap/sssd_ldap_configure_tls_reqcert/rule.yml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    disa: CCI-001453
+    nist: SC-12(3),CM-6(a)
+    srg: SRG-OS-000250-GPOS-00093
+    stigid@rhel7: RHEL-07-040190
+
+ocil_clause: 'the TLS reqcert is not set to demand'
+
+ocil: |-
+    To verify the LDAP client backend demands a valid certificate from the server in
+    remote ldap access sessions, run the following command:
```
[the remark this answers is not in the document]
Comment (ggbecker) [reply linux_os/guide/services/sssd/sssd-ldap/sssd_ldap_configure_tls_reqcert/rule.yml]: Done

**Comment by ggbecker** [comment]
> > @ggbecker Do you have any plans?
> 
> Working on that right now. Thanks

Should have everything addressed right now.

**Comment by mildas** [comment]
**Changes identified:**
Rule sssd_ldap_configure_tls_reqcert:
 Bash remediation is newly added.
 Ansible remediation newly added.
 OVAL check is newly added.
Profile stig on rhel7:
 Rule sssd_ldap_configure_tls_reqcert added to stig profile.

**Recommended tests to execute:**
 build_product rhel7
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using ansible --datastream build/ssg-rhel7-ds.xml sssd_ldap_configure_tls_reqcert
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-rhel7-ds.xml sssd_ldap_configure_tls_reqcert
 tests/test_suite.py profile --libvirt qemu:///system test-suite-vm --datastream build/ssg-rhel7-ds.xml stig

**Comment by mildas** [comment]
**Changes identified:**
Rule sssd_ldap_configure_tls_reqcert:
 Ansible remediation newly added.
 Bash remediation is newly added.
 OVAL check is newly added.
Profile stig on rhel7:
 Rule sssd_ldap_configure_tls_reqcert added to stig profile.

**Recommended tests to execute:**
 build_product rhel7
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using ansible --datastream build/ssg-rhel7-ds.xml sssd_ldap_configure_tls_reqcert
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-rhel7-ds.xml sssd_ldap_configure_tls_reqcert
 tests/test_suite.py profile --libvirt qemu:///system test-suite-vm --datastream build/ssg-rhel7-ds.xml stig

**Pull request merged**